### PR TITLE
[hotfix/JENKINS-38345] REGRESSION remove classList

### PR DESCRIPTION
**Pull request by scherler**
# Description

See [JENKINS-38345](https://issues.jenkins-ci.org/browse/JENKINS-38345).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@jenkinsci/code-reviewers @reviewbybees

### CODE REVIEW

**Comment by sophistifunk** [comment]
Looks fine 
bbm

File: blueocean-dashboard/src/main/js/LoadingIndicator.js [hunk 1]
```diff
@@ -49,7 +49,14 @@ export default {
             clearTimeouts();
             setLoaderClass('complete', 10);
             timeouts.push(setTimeout(() => {
-                document.getElementById('loadbar').classList = '';
+                // The Element.classList is a read-only property
+                const classList = document.getElementById('loadbar').classList;
+                var classListAsArray = new Array(classList.length);
```
Comment (tfennelly) [inline blueocean-dashboard/src/main/js/LoadingIndicator.js]: Maybe wrap this in `if (classList)` in case the property doesn't exist at all.

Comment (sophistifunk) [inline blueocean-dashboard/src/main/js/LoadingIndicator.js]: classList is `[]` (well, close enough to it) when empty

File: blueocean-dashboard/src/main/js/LoadingIndicator.js [hunk 2]
```diff
@@ -49,7 +49,14 @@ export default {
             clearTimeouts();
             setLoaderClass('complete', 10);
             timeouts.push(setTimeout(() => {
-                document.getElementById('loadbar').classList = '';
+                // The Element.classList is a read-only property
+                const classList = document.getElementById('loadbar').classList;
+                const classListAsArray = new Array(classList.length);
+                for (let i = 0, len = classList.length; i < len; i++) {
+                    classListAsArray[i] = classList[i];
```
Comment (kzantow) [inline blueocean-dashboard/src/main/js/LoadingIndicator.js]: why not just classList.remove here?

File: blueocean-dashboard/src/main/js/LoadingIndicator.js [hunk 3]
```diff
@@ -49,7 +49,14 @@ export default {
             clearTimeouts();
             setLoaderClass('complete', 10);
             timeouts.push(setTimeout(() => {
-                document.getElementById('loadbar').classList = '';
```
Comment (kzantow) [inline blueocean-dashboard/src/main/js/LoadingIndicator.js]: this was a fat finger. it should have been `...className = '';`

**Comment by kzantow** [comment]
Side note, this isn't a regression, just a bug.

File: blueocean-dashboard/src/main/js/LoadingIndicator.js (hunk 1 above)
Comment (scherler) [inline blueocean-dashboard/src/main/js/LoadingIndicator.js]: yeah you are right, will adopt that

**Pull request merged**